doc/design/GeneratedClientDesign.md

```diff
@@ -0,0 +1,133 @@
+# Generated Client Design
+
+## Introduction
+
+This document tracks design rules and decisions for .NET clients generated by the DPG generator.  It may later be extended to include other generator designs.
+
+<!--
+## Clients
+
+TBD
+
+## Client Protocol Operations
+
+TBD
+
+## Client Convenience Operations
+-->
+
+## Model Generation
+
+Models are defined as partial classes with one or more constructors, properties, and methods that read from/write to wire formats.
+
+### Model Namespace
+
+Generated models will be defined in a `.Models` namespace.
```

Comment:
Not having the namespace is the most common case, to use as a default.

Do other languages care?  C# only?
@tg-msft, can you chime in with your thoughts? @KrzysztofCwalina to drive closure.



Only add this if specified in an API review.

Will need Cadl support for specifying to use the namespace.

Reply:
Java ALWAYS wants a `.Models` even if it's only got one type.  TS has never done `-models` because they've never had great namespacing options ([though that might be changing?](https://gist.github.com/mpodwysocki/321a24404bd38be4980c331d8da9537c)).  Python uses `-models` sometimes, but not too often from what I can tell.

I would personally lean toward `.Models` as the default for .NET because it leaves more room for growth.  We should explicitly choose not to have a namespace only if we've discussed it with a service and they have high confidence there won't be significant expansion.

Reply:
Filed https://github.com/Azure/cadl-azure/issues/1787 to track the ask on the Cadl side.